src/Arrays/MPIStateArrays.jl

```diff
@@ -856,4 +858,43 @@ end
     end
 end
 
+"""
+    throw_not_finite_error(Q::MPIStateArray)
+
+Call to throw an error if `isfinite(norm(Q)) == false`
+to provide a detailed error of which field(s) are not
+finite.
+
+!!! warn
+    This is an expensive method, which calls `norm(Q)`
+"""
+function throw_not_finite_error(Q::MPIStateArray)
+    normQ = norm(Q)
```

Comment:
I'm confused -- why call `norm` again here? We always call it in the callback, since we output it, and we only call this function if `isnan(norm(Q))`. If you want to make that `isfinite` instead of `isnan` in the caller, that's fine, but IMO this function can always assume that the `norm` is not right and will inspect the fields to provide more detail -- the name of the function makes it clear that it will throw an error.

Reply:
If we don't call `norm`, then we don't know for sure that `all(isfinite.(norm(Q)))` and that complicated what we'd write for the error message (what error should we throw if `all(isfinite.(norm(Q)))`?). Maybe we can change the function signature to
```julia
throw_not_finite_error(Q::MPIStateArray, normQ = norm(Q))```
? and make sure that we pass the norm in?

Reply:
What about throwing the error with nothing in it? I agree with @kpamnany though, the call makes it clear that an error should be thrown, and if you do not throw the error that's weird.

Reply:
I think it's perfectly fine to assume an entering condition of `!isfinite(norm(Q))` as long as it's documented. Someone is calling a function called `throw_not_finite_error` after all.

If you don't like that then passing in the norm is better than calling `norm` again.

Reply:
Another option is to just print a warning and let the user decide what they want to do afterward.

Reply:
(I mean in all cases, and never throw an error. Make it more diagnostic, not fatal.)

Reply:
> (I mean in all cases, and never throw an error. Make it more diagnostic, not fatal.)

I like this idea. Make this something like `show_not_finite_fields` which shows nothing if they're all finite. Then throw the error from the callback as before.

Reply:
Ahh, that's a great idea. Updating...